[Bug]: Failing test 1

### Brief problem description

test 1 fails 
FAILED tests/test_items.py::test_get_item_1 - assert 404 == 200

### Steps to Reproduce

run the tests 

### Expected Result

all test passed 

### Actual Result

FAILED tests/test_items.py::test_get_item_1 - assert 404 == 200

--- a/src/app/services/item_service.py
+++ b/src/app/services/item_service.py
@@ -228,8 +228,11 @@ def get_item_by_id_dfs_iterative(
 
                     for task in lab.tasks:
                         counter += 1
-                        if lab.id == item_id:
+                        # --- ИСПРАВЛЕНИЕ ЗДЕСЬ ---
+                        # Было: if lab.id == item_id:
+                        if task.id == item_id:
                             return FoundItem(task, counter)
+                        # -------------------------
 
                         for step in task.steps:
                             counter += 1

